system/Database/Postgre/Builder.php

```diff
@@ -364,6 +364,12 @@ protected function _upsertBatch(string $table, array $keys, array $values): stri
                 }
             }
 
+            $alias = $this->QBOptions['alias'] ?? '"excluded"';
+
+            if (strtolower($alias) !== '"excluded"' && $this->db->DBDebug) {
+                throw new DatabaseException('Postgres alias is always named "excluded". A custom alias cannot be used.');
+            }
```

Comment:
I guess it doesn't work if `$this->db->DBDebug` is false and using a custom alias?

Reply:
Honestly I'm not quite sure what the reasons are for using this. I was just following the pattern that I saw being used in the rest of the code. Can you elaborate on why this should or should not be used?

Reply:
I guess this is just an extra layer of protection to prevent sensitive database errors from being displayed. ?? In this case if the error is not thrown the script will go on to throw an error with the query which could then potentially display sensitive info so maybe its best to throw regardless of debug mode.

Reply:
Removed DBDebug test.

Reply:
This is because of historical reasons.

1. The `DBDebug` option was originally a debugging option that decides whether or not database errors should be displayed on the screen. So if `DBDebug` is false, CI should return `false` without throwing an exception.
   > db_debug | TRUE/FALSE (boolean) - Whether database errors should be displayed. https://codeigniter.com/userguide3/database/configuration.html#explanation-of-values
2. Before v4.3.0, in production environment `DBDebug` is `false` by default. https://github.com/codeigniter4/CodeIgniter4/blob/020af6d2d2e61dca96797240308bb96bb080a77a/app/Config/Database.php#L42
3. But it is not good practice that changing the behavior between production and development environment. So we have changed `DBDebug` to `true` anytime by default since v4.3.0. See https://github.com/codeigniter4/CodeIgniter4/blob/4.3/user_guide_src/source/changelogs/v4.3.0.rst#changes
4. As a result, now `DBDebug` means whether or not to throw an exception when an database error occurs.

Reply:
> I guess this is just an extra layer of protection to prevent sensitive database errors from being displayed. 

I also think the `DBDebug` option was originally created like that.
But displaying database errors has nothing to do with the option now.
Because CI4 never displays error messages on production environment.

However, if a database error occurs and `DBDebug` is `false`, methods must be implemented to return `false` without throwing an exception.

Reply:
I think `DatabaseException` is a runtime exception that is thrown when a database query fails.

In this case, it seems better to use `InvalidArgumentException` or `BadMethodCallException` that is a logic exception. Logic Exception represents error in the program logic. We don't need to check `DBDebug`.

Reply:
I would like to remove `DBDebug` in the future. Because it seems useless.
But it is a big breaking change. So it will take long time.
Until then, we need to implement database methods to return `false` when `DBDebug` is `false`.

Reply:
Changed to use `CodeIgniter\Database\InvalidArgumentException`